commands/mr/update/mr_update.go

```diff
@@ -70,9 +71,16 @@ func NewCmdUpdate(f *cmdutils.Factory) *cobra.Command {
 			}
 
 			l.Title = gitlab.String(mergeTitle)
+
+			if cmd.Flags().Changed("lock-discussion") && cmd.Flags().Changed("unlock-discussion") {
+				return &cmdutils.FlagError{Err: errors.New("--lock-discussion and --unlock-discussion can't be used together")}
+			}
```

Comment:
The check is done too late when most operations have been done. The user shouldn't be delayed. You could put it somewhere up above this:
```go
apiClient, err := f.HttpClient()
```